tests/test_datasets.py

```diff
@@ -747,5 +747,39 @@ def test_scanpaths_from_fixations(fixation_indices):
     compare_fixations(sub_fixations, new_sub_fixations, crop_length=True)
 
 
+invert_param_values = [True, False]
+@pytest.mark.parametrize('attribute_name , attribute_value', [('dva',1), ('dva',4), ('dva',15), ('some_strings','a'), ('some_strings','n'), ('some_strings','q')])
+@pytest.mark.parametrize('invert_match', invert_param_values)
+def test_filter_stimuli_by_attribute(file_stimuli_with_attributes, fixation_trains, attribute_name, attribute_value, invert_match):
+
+    fixations = fixation_trains[np.arange(len(fixation_trains))]
+    # Access the attribute using the attributes dictionary
+    attribute_data = file_stimuli_with_attributes.attributes[attribute_name]
+    mask = np.array([element == attribute_value for element in attribute_data])
```

Comment:
It's nice that you're testing multiple values via parametrize, but that means that you're essentially writing a regression test against the current implementation, which is copied here. I think it might be better to test only one or two values, and for those values handcode the correct indices (there are not that many), to check that it's actually the right ones. Right now, in theory your code could always match everything and the test would pass